common/lib/xmodule/xmodule/js/src/capa/display.coffee

```diff
@@ -527,12 +541,14 @@ class @Problem
         answered = false
       if bind
         $(select_field).on 'change', (e) =>
-          @submitAnswersAndSubmitButton()
+          @submitAnswersAndSubmitButton false, true
           return
         return
 
     if answered
       @enableSubmitButton true
+      if data_changed
+        @saveNotification.remove()
```

Comment:
If this check is not there then the save notification will be removed just because the question is answered.  This check looks for an answer and a detected change.